app/src/features/dataset_explorer/DatasetExplorer.tsx

```diff
@@ -1,76 +1,61 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
 import DatasetFilter from "./DatasetFilter";
-import DataFetcher from "../../utils/DataFetcher";
 import DataTable from "./DataTable";
 import DatasetListing from "./DatasetListing";
 import styles from "./DatasetExplorer.module.scss";
 import ChevronRightIcon from "@material-ui/icons/ChevronRight";
-import { DatasetMetadata } from "../../utils/DatasetMetadata";
-
-type LoadStatus = "unloaded" | "loading" | "loaded";
+import { useDatasetStore } from "../../utils/useDatasets";
+import { DatasetStore } from "../../utils/DatasetTypes";
 
 function renderTableOrPlaceholder(
```

Comment:
nit: nlow that this page also has fields table, maybe we should update this to renderPreviewTableOrPlaceholder or even just function PreviewTable. Also, is it not technically its own component? Can it get DatasetStore directly instead of params?

Reply:
Good call on both accounts. I moved this to its own file. I also extracted the "select a dataset to view" so that the data preview can be more portable and used in another context